### PR TITLE
NR-217272: add ec2 filters + prefix + default instances

### PR DESCRIPTION
**Pull request by rogercoll**
- TODO: add windows instances

### CODE REVIEW

File: terraform/otel-ec2/main.tf [hunk 1]
```diff
@@ -1,9 +1,18 @@
+locals {
+    // filter EC2 instances to launch
+    filtered_ec2_agents = length(var.ec2_filters) == 0 ? var.ec2_otels : {
+    for k, v in var.ec2_otels : k => v if anytrue([for f in var.ec2_filters :
+        strcontains(k, f)]) }
+
+    // Replace the TAG_OR_UNIQUE_NAME
+    assembled_ec2 = var.ec2_prefix == "" ? local.filtered_ec2_agents : { for k, v in local.filtered_ec2_agents : format("%s-%s", var.ec2_prefix, k) => v }
```
Comment (rubenruizdegauna) [inline terraform/otel-ec2/main.tf:8]: Right now, we are using `%s:%s`  for prefix, aren't we?

File: terraform/otel-ec2/vars.tf [hunk 2]
```diff
@@ -51,3 +47,254 @@ variable "inventory_output" {
...
+  description = "EC2 instances names prefix, it will appended to the ec2_otels map key"
+  default = ""
+}
+
+variable "ec2_filters" {
+  description = "EC2 instances names to deploy"
+  default = []
+}
+
+variable "ec2_otels" {
```
Comment (rubenruizdegauna) [inline terraform/otel-ec2/vars.tf:61]: In the Infra Agent we have 44 AMIs. Why only 18 here? 👇

Comment (rogercoll) [reply terraform/otel-ec2/vars.tf:61]: Good point, I was focused on the SA as it is not a breaking change. I can add infra agent's ones too

File: terraform/otel-ec2/main.tf (hunk 1 above)
Comment (rogercoll) [reply terraform/otel-ec2/main.tf]: Fixed in https://github.com/newrelic-experimental/env-provisioner/pull/18/commits/104fe56e3c12f284e0a30049eea22e2cd7bc63ab

File: terraform/otel-ec2/vars.tf (hunk 2 above)
Comment (rubenruizdegauna) [reply terraform/otel-ec2/vars.tf:61]: Sounds good. I'm finishing with packaging tests in the Infra Agent. Once is merged, I can add the rest.

**Review by rubenruizdegauna — APPROVED** [review]
LGTM

**Pull request merged**